platform/viewer/src/lib/DICOMSR/retrieveDataFromSR.js

```diff
@@ -0,0 +1,71 @@
+import * as dcmjs from 'dcmjs';
+import { getAllDisplaySets, getInstanceMetadata } from './srUtils';
+
+const retrieveDataFromSR = Part10SRArrayBuffer => {
+  const allDisplaySets = getAllDisplaySets();
+
+  // Get the dicom data as an Object
+  const dicomData = dcmjs.data.DicomMessage.readFile(Part10SRArrayBuffer);
+  const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
+    dicomData.dict
+  );
+
+  // Convert the SR into the kind of object the Measurements package is expecting
+  return imagingMeasurementsToMeasurementData(dataset, allDisplaySets);
+};
+
+const imagingMeasurementsToMeasurementData = (dataset, displaySets) => {
+  const { MeasurementReport } = dcmjs.adapters.Cornerstone;
+  const storedMeasurementByToolType = MeasurementReport.generateToolState(
+    dataset
+  );
+  const measurementData = {};
+  let measurementNumber = 0;
+
```

Comment:
This could be a good place to allocate different colors for each series and then you can distinguish between measurements made at different times and the ones in the current session. Once a color is chosen it can be set like ```color: "someCssColor"``` around line 52 below in the toolData.

Reply:
You are right. Althought I've talked with Erik and for now we want to have only one SR being used, the latest one.